analytic_engine/src/sst/parquet/async_reader.rs

```diff
@@ -540,10 +540,21 @@ impl Stream for RecordBatchProjector {
                         }
                         projector.metrics.row_num += record_batch.num_rows();
 
-                        let projected_batch =
-                            FetchedRecordBatch::try_new(&projector.row_projector, record_batch)
-                                .box_err()
-                                .context(DecodeRecordBatch {});
+                        let fetched_schema = projector.row_projector.fetched_schema().clone();
+                        let primary_key_indexes = projector
+                            .row_projector
+                            .primary_key_indexes()
+                            .map(|idxs| idxs.to_vec());
```

Comment:
There is no clone before, why do clone after this change set?

Reply:
It has actually... but happen in `FetchedRecordBatch::try_new`